recipes/autoconf/all/conanfile.py

```diff
@@ -13,110 +18,86 @@ class AutoconfConan(ConanFile):
-                  destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def build(self):
+        apply_conandata_patches(self)
+        autotool = Autotools(self)
+        autotool.configure()
+        autotool.make()
+        autotool.install(args=[f"DESTDIR={unix_path(self, self.package_folder)}"])
+
+    def package(self):
+        packager = AutoPackager(self)
+        packager.run()
+        rmdir(self, Path(self.package_folder, "share", "info"))
+        rmdir(self, Path(self.package_folder, "share", "man"))
 
     @property
     def _datarootdir(self):
-        return os.path.join(self.package_folder, "bin", "share")
+        return Path(self.package_folder, "share")
 
     @property
     def _autoconf_datarootdir(self):
-        return os.path.join(self._datarootdir, "autoconf")
-
-    def _configure_autotools(self):
-        if self._autotools:
-            return self._autotools
-        self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
-        datarootdir = self._datarootdir
-        prefix = self.package_folder
-        if self.settings.os == "Windows":
-            datarootdir = tools.unix_path(datarootdir)
-            prefix = tools.unix_path(prefix)
-        conf_args = [
-            "--datarootdir={}".format(datarootdir),
-            "--prefix={}".format(prefix),
-        ]
-        self._autotools.configure(args=conf_args, configure_dir=self._source_subfolder)
-        return self._autotools
-
-    def _patch_files(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
-
-    @contextlib.contextmanager
-    def _build_context(self):
-        with tools.environment_append(tools.RunEnvironment(self).vars):
-            yield
-
-    def build(self):
-        self._patch_files()
-        with self._build_context():
-            autotools = self._configure_autotools()
-            autotools.make()
-
-    def package(self):
-        self.copy("COPYING*", src=self._source_subfolder, dst="licenses")
-        with self._build_context():
-            autotools = self._configure_autotools()
-            autotools.install()
-        tools.rmdir(os.path.join(self.package_folder, "bin", "share", "info"))
-        tools.rmdir(os.path.join(self.package_folder, "bin", "share", "man"))
+        return Path(self._datarootdir, "autoconf")
 
     def package_info(self):
-        self.cpp_info.libdirs = []
-        self.cpp_info.includedirs = []
-
-        bin_path = os.path.join(self.package_folder, "bin")
-        self.output.info("Appending PATH env var with : {}".format(bin_path))
-        self.env_info.PATH.append(bin_path)
+        bin_path = Path(self.package_folder, "bin")
+        self.output.info(f"Appending PATH env var with : {bin_path}")
+        self.buildenv_info.prepend_path("PATH", str(bin_path))
```

Comment:
```suggestion
```

Useless. `bindirs` of tool_requires are automatically injected to PATH in build scope of conan v2 recipes.